source/blender/gpu/intern/gpu_material.c

```diff
@@ -877,22 +883,81 @@ static void shade_light_textures(GPUMaterial *mat, GPULamp *lamp, GPUNodeLink **
 
 		if (mtex && mtex->tex->type & TEX_IMAGE && mtex->tex->ima) {
-			mat->dynproperty |= DYN_LAMP_PERSMAT;
+			if (lamp->type == LA_AREA) {
+				mat->dynproperty = DYN_LAMP_VEC | DYN_LAMP_CO | DYN_LAMP_AREAMAT;
+			}
+			else {
+				mat->dynproperty |= DYN_LAMP_PERSMAT;
+			}
 
 			float one = 1.0f;
 			GPUNodeLink *tex_rgb;
 
 			GPU_link(mat, "shade_light_texture",
-			         GPU_builtin(GPU_VIEW_POSITION),
-			         GPU_image(mtex->tex->ima, &mtex->tex->iuser, false), GPU_uniform(mtex->size),
+					 GPU_builtin(GPU_VIEW_POSITION),
+					 GPU_image(mtex->tex->ima, &mtex->tex->iuser, false), GPU_uniform(mtex->size),
 					 GPU_select_uniform(&mtex->lodbias, GPU_DYNAMIC_TEX_LODBIAS, NULL, mat->ma),
-			         GPU_dynamic_uniform((float *)lamp->dynpersmat, GPU_DYNAMIC_LAMP_DYNPERSMAT, lamp->ob),
-			         &tex_rgb);
+					 GPU_dynamic_uniform((float *)lamp->dynpersmat, GPU_DYNAMIC_LAMP_DYNPERSMAT, lamp->ob), &tex_rgb);
+
 			texture_rgb_blend(mat, tex_rgb, *rgb, GPU_uniform(&one), GPU_uniform(&mtex->colfac), mtex->blendtype, rgb);
 		}
 	}
 }
 
+static void shade_area_diff_texture(GPUMaterial *mat, GPULamp *lamp, GPUNodeLink **rgb, GPUShadeInput *shi)
+{
+	GPUNodeLink * tex_rgb;
+	MTex *mtex = NULL;
+	int i;
+	float one = 1.0f;
+
+	for (i = 0; i < MAX_MTEX; ++i) {
+		mtex = lamp->la->mtex[i];
+     
+		if (mtex && mtex->tex->type & TEX_IMAGE && mtex->tex->ima) {
+			mat->dynproperty |= DYN_LAMP_CO | DYN_LAMP_AREAMAT;
+			GPU_link(mat, "area_diff_texture",
+					 GPU_builtin(GPU_VIEW_POSITION),
+					 GPU_image(mtex->tex->ima, &mtex->tex->iuser, false),
+					 GPU_select_uniform(&mtex->lodbias, GPU_DYNAMIC_TEX_LODBIAS, NULL, mat->ma),
+					 GPU_dynamic_uniform(lamp->dynco, GPU_DYNAMIC_LAMP_DYNCO, lamp->ob),
+					 shi->vn,
+					 GPU_dynamic_uniform((float *)lamp->dynareamat, GPU_DYNAMIC_LAMP_DYNAREAMAT, lamp->ob),
+					 GPU_dynamic_uniform((float *)lamp->area_size, GPU_DYNAMIC_LAMP_DYNAREASIZE, lamp->ob),
+					 shi->har,
+					 &tex_rgb);
+			texture_rgb_blend(mat, tex_rgb, *rgb, GPU_uniform(&one), GPU_uniform(&mtex->colfac), MTEX_MUL, rgb);
+		}
+	}
+}
+
+static void shade_area_spec_texture(GPUMaterial *mat, GPULamp *lamp, GPUNodeLink **rgb, GPUShadeInput *shi)
+{
+	GPUNodeLink *tex_rgb;
+	MTex *mtex = NULL;
+	int i;
+	float one = 1.0f;
+
+	for (i = 0; i < MAX_MTEX; ++i) {
+		mtex = lamp->la->mtex[i];
+
+		if (mtex && mtex->tex->type & TEX_IMAGE && mtex->tex->ima) {
+			mat->dynproperty |= DYN_LAMP_CO | DYN_LAMP_AREAMAT;
+			GPU_link(mat, "area_spec_texture",
+					 GPU_builtin(GPU_VIEW_POSITION),
+					 GPU_image(mtex->tex->ima, &mtex->tex->iuser, false),
+					 GPU_select_uniform(&mtex->lodbias, GPU_DYNAMIC_TEX_LODBIAS, NULL, mat->ma),
+					 GPU_dynamic_uniform(lamp->dynco, GPU_DYNAMIC_LAMP_DYNCO, lamp->ob),
+					 shi->vn,
+					 GPU_dynamic_uniform((float *)lamp->dynareamat, GPU_DYNAMIC_LAMP_DYNAREAMAT, lamp->ob),
+					 GPU_dynamic_uniform((float *)lamp->area_size, GPU_DYNAMIC_LAMP_DYNAREASIZE, lamp->ob),
```

Comment:
Sometimes when we play with render to texture, the image is not centered correctly. An option that could be cool to have is the offset buttons (beside texture size). I'll do some tests... Only as GPU_uniforms